grpcutil/status.go

```
@@ -60,6 +61,19 @@ func ErrorToStatusCode(err error) codes.Code {
 	return codes.Unknown
 }
 
+// Status creates a new a *github.com/gogo/status.Status with the
+// given error code, error message and error details.
+func Status(errCode codes.Code, errMessage string, details ...proto.Message) *status.Status {
+	stat := status.New(errCode, errMessage)
+	if len(details) > 0 {
+		statWithDetails, err := stat.WithDetails(details...)
+		if err == nil {
+			return statWithDetails
+		}
```

This silently ignores the error if `WithDetails` fails. When can it fail?

---

Internally, `status.Status.WithDetails()` when it calls `types.MarshalAny(detail)` on each of the provided details.

In that case, we just return `stat`, i.e., the `status.Status` object without details.

---

Should we panic in that case? We should handle the error somehow: if it can happen, we should return it to the caller, if it can't, we should panic.

---

I will rework this implementation...
I am quite sure we have a similar code in Mimir too, where this error is silently ignored. But this is a separate story

---

I have changed this implementation @colega